app-android/src/main/java/com/ustadmobile/port/android/view/binding/TextViewBindings.kt

```diff
@@ -204,3 +206,66 @@ fun TextView.setHtmlText(htmlText: String?) {
 fun TextView.setFileSize(fileSize: Long) {
     text = UMFileUtil.formatFileSize(fileSize)
 }
+
+@BindingAdapter(value=["clazzMemberWithClazzWorkAndProgress"])
+fun TextView.setClazzWorkMarking(clazzMemberWithClazzWorkAndProgress: ClazzMemberWithClazzWorkProgress){
+    val marked = context.getString(R.string.marked_cap)
+    val completed = context.getString(R.string.completed)
+    val submitted = context.getString(R.string.submitted_cap)
+    val notSubmitted = context.getString(R.string.not_submitted_cap)
+    val ofContent = context.getString(R.string.of_content)
+
+    val clazzWorkSubmission = clazzMemberWithClazzWorkAndProgress.mClazzWorkSubmission
+    var line2Bit1 = ""
+    var line2Bit2 = ""
+    if(clazzWorkSubmission == null){
+        line2Bit1 = notSubmitted
+    }else{
+        line2Bit1 = if(clazzWorkSubmission.clazzWorkSubmissionDateTimeFinished > 0){
+            if(clazzWorkSubmission.clazzWorkSubmissionDateTimeMarked > 0){
+                marked
+            }else{
+                submitted
+            }
+        }else{
+            notSubmitted
+        }
+    }
+
+    if(clazzMemberWithClazzWorkAndProgress.clazzWorkHasContent &&
+            clazzMemberWithClazzWorkAndProgress.mProgress >= 0){
+        line2Bit2 = " - " + completed + " " + clazzMemberWithClazzWorkAndProgress.mProgress.toInt() +
+                "% " + ofContent
+    }
+    val line2 = line2Bit1 + line2Bit2
+
+    text = line2
+}
+
+@BindingAdapter(value=["selectedClazzWorkQuestionType"])
+fun TextView.setTypeText(clazzWorkQuestionType: Int){
+    if(clazzWorkQuestionType == ClazzWorkQuestion.CLAZZ_WORK_QUESTION_TYPE_FREE_TEXT){
+        text = context.getString(R.string.sel_question_type_free_text)
+    }else if(clazzWorkQuestionType == ClazzWorkQuestion.CLAZZ_WORK_QUESTION_TYPE_MULTIPLE_CHOICE){
+        text = context.getString(R.string.quiz)
+    }
+}
+
+
+@BindingAdapter(value=["responseTextFilled"])
+fun TextView.setResponseTextFilled(responseText: String?){
+    if(responseText == null || responseText.isEmpty()){
+        text = context.getString(R.string.not_answered)
+    }else{
+        text = responseText
+    }
+}
+
+@BindingAdapter(value=["submissionButtonText"])
+fun Button.setSubmissionButtonText(more: Boolean){
```

Comment:
Couldn't this be done in the layout xml itself using a conditional operator? E.g.

```
<Button 
   android:text="@{more ? @string/return_and_mark_next : @string/return_only}"
```